### PR TITLE
[GG-2583] Add popups to both poi and state maps

### PR DESCRIPTION
**Pull request by sshalu12**
Add popups to both poi and state maps

### CODE REVIEW

File: app/src/poi.js [hunk 1]
```diff
@@ -31,6 +31,42 @@ export default function Poi() {
...
+          e.features[0].properties.state +
+          ", " +
+          e.features[0].properties.zip +
+          ", " +
+          e.features[0].properties.country;
+        const category =
+          e.features[0].properties.category_name +
+          " (" +
+          e.features[0].properties.category_id +
+          ") ";
```
Comment (ashutoshgt) [inline app/src/poi.js]: This can be improved by using string formatting in JS. Check the usage of `{}` in strings in JS.

File: app/src/poi.js [hunk 2]
```diff
@@ -31,6 +31,42 @@ export default function Poi() {
...
+      const popup = new mapboxgl.Popup({
+        closeButton: false,
+        closeOnClick: false,
+      });
+      map.current.on("mousemove", "points-of-interest", (e) => {
+        // Change the cursor to a pointer when the mouse is move over the points-of-interest layer.
+        map.current.getCanvas().style.cursor = "pointer";
+        // Copy coordinates array.
+        const coordinates = e.features[0].geometry.coordinates.slice();
+        const name = e.features[0].properties.business_name;
```
Comment (sushilsingh95) [inline app/src/poi.js]: Do we need to traverse this entire property chain `e.features[0].properties` everytime. Can we store it in a variable.

**Review by ashutoshgt — CHANGES_REQUESTED** [review]

File: app/src/poi.js [hunk 3]
```diff
@@ -31,6 +31,42 @@ export default function Poi() {
...
+          e.features[0].properties.country;
+        const category =
+          e.features[0].properties.category_name +
+          " (" +
+          e.features[0].properties.category_id +
+          ") ";
+        popup
+          .setLngLat(coordinates)
+          .setHTML("<p>" + name + "<br>" + address + "<br>" + category + "</p>")
+          .addTo(map.current);
```
Comment (sushilsingh95) [inline app/src/poi.js:50]: before adding it to map, should we check whether it already exists or not?

**Review by sushilsingh95 — CHANGES_REQUESTED** [review]

File: app/src/poi.js (hunk 3 above)
Comment (sshalu12) [reply app/src/poi.js:50]: https://github.com/mapbox/mapbox-gl-js/blob/16756b331d8d0d8aa63ef02bccdef77d748ba3a8/src/ui/marker.js#L189

 line number 189 of this js file. 
.addTo Don't add data in map again.

File: app/src/poi.js [hunk 4]
```diff
@@ -31,6 +31,30 @@ export default function Poi() {
...
+        closeOnClick: false,
+      });
+      map.current.on("mousemove", "points-of-interest", (e) => {
+        // Change the cursor to a pointer when the mouse is move over the points-of-interest layer.
+        map.current.getCanvas().style.cursor = "pointer";
+        // Copy coordinates array.
+        const coordinates = e.features[0].geometry.coordinates.slice();
+        const address_data = e.features[0].properties;
+        const name = address_data["business_name"];
+        const address = `${address_data["address"]} , ${address_data["city"]} , ${address_data["state"]} , ${address_data["zip"]} , ${address_data["country"]}`;
```
Comment (ashutoshgt) [inline app/src/poi.js]:
```suggestion
        const address = `${address_data["address"]}, ${address_data["city"]}, ${address_data["state"]}, ${address_data["zip"]}, ${address_data["country"]}`;
```

File: app/src/poi.js [hunk 5]
```diff
@@ -31,6 +31,30 @@ export default function Poi() {
...
+        const category = `${address_data["category_name"]} (${address_data["category_id"]})}`;
```
Comment (ashutoshgt) [inline app/src/poi.js]:
```suggestion
        const category = `${address_data["category_name"]} (${address_data["category_id"]})`;
```

File: app/src/poi.js [hunk 6]
```diff
@@ -31,6 +31,30 @@ export default function Poi() {
...
+        closeButton: false,
+        closeOnClick: false,
+      });
+      map.current.on("mousemove", "points-of-interest", (e) => {
+        // Change the cursor to a pointer when the mouse is move over the points-of-interest layer.
+        map.current.getCanvas().style.cursor = "pointer";
+        // Copy coordinates array.
+        const coordinates = e.features[0].geometry.coordinates.slice();
+        const address_data = e.features[0].properties;
+        const name = address_data["business_name"];
```
Comment (ashutoshgt) [inline app/src/poi.js]:
```suggestion
        const name = address_data.business_name;
```
Please use dot notation everywhere.

**Review by ashutoshgt — CHANGES_REQUESTED** [review]

File: app/src/poi.js [hunk 7]
```diff
@@ -31,6 +31,30 @@ export default function Poi() {
...
+      const popup = new mapboxgl.Popup({
+        closeButton: false,
+        closeOnClick: false,
+      });
+      map.current.on("mousemove", "points-of-interest", (e) => {
+        // Change the cursor to a pointer when the mouse is move over the points-of-interest layer.
+        map.current.getCanvas().style.cursor = "pointer";
+        // Copy coordinates array.
+        const coordinates = e.features[0].geometry.coordinates.slice();
+        const address_data = e.features[0].properties;
```
Comment (ashutoshgt) [inline app/src/poi.js]:
```suggestion
        const locationData = e.features[0].properties;
```
In JS we use `camelCase`

File: app/src/poi.js (hunk 3 above)
Comment (sushilsingh95) [reply app/src/poi.js:50]: great! then we don't need to do this.

**Review by ashutoshgt — APPROVED** [review]

**Pull request merged**